site/content/blog/2020-01-08-incorporating-golangci-lint.md

```diff
@@ -0,0 +1,140 @@
+---
+title: Incorporating GolangCI-Lint at Mattermost
+slug: incorporating-golangci-lint
+date: 2020-01-08
+categories:
+    - "go"
+    - “linting”
+author: Agniva De Sarker
+github: agnivade
+community: agnivade
+author_2: Ben Schumacher
+github_2: hanzei
+community_2: hanzei
+---
+
+
+At Mattermost, we have traditionally relied on the trusty `go vet` and `gofmt` checks for our CI runs. While that is great, there were a lot of other powerful linters out there which we were missing out on.
+
+Speaking of linters, the first name that inevitably comes up is [staticcheck](https://staticcheck.io/). It is a powerful metalinter with a whole slew of checks. But simply running staticcheck was not sufficient, because it missed out on other linters which did one little thing, but nevertheless were very powerful. A few popular ones are [ineffassign](https://github.com/gordonklaus/ineffassign), [errcheck](https://github.com/kisielk/errcheck), [unconvert](https://github.com/mdempsky/unconvert), [deadcode](https://github.com/remyoudompheng/go-misc/tree/master/deadcode), [structcheck](https://gitlab.com/opennota/check/tree/master/cmd/structcheck) amongst others.
```

Comment:
```suggestion
Speaking of linters, the first name that inevitably comes up is [staticcheck](https://staticcheck.io/). It's a powerful metalinter with a whole slew of checks. But simply running staticcheck is not sufficient, because it misses out on other linters which perform a single task, but nevertheless are very powerful. A few popular ones are [ineffassign](https://github.com/gordonklaus/ineffassign), [errcheck](https://github.com/kisielk/errcheck), [unconvert](https://github.com/mdempsky/unconvert), [deadcode](https://github.com/remyoudompheng/go-misc/tree/master/deadcode), and [structcheck](https://gitlab.com/opennota/check/tree/master/cmd/structcheck) among others.
```